test/unit/nodes/element.js

```diff
@@ -18,6 +18,23 @@ suite('Element', function() {
 			);
 		});
 
+		suite('#querySelectorAll', function() {
+			test('css selector with tag name only', function() {
+				var lis = elem.querySelectorAll('li');
```

Comment:
You are right. Interestingly, this indeed seems to reveal a bug in querySelectorAll(). Even `elem.querySelectorAll('ul li')` returns an empty array.
I'll investigate further when I have more time.